### ISSUE TITLE
[apex] ApexCRUDViolation only works for a single method in a class

### ISSUE THREAD

**Issue by dpr-nh** [pmd/pmd#4589]
**Affects PMD Version:** 
6.55.0

**Rule:**

[ApexCRUDViolation](https://docs.pmd-code.org/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**
The ApexCRUDViolation rule is not working properly for multiple methods in a class

**Code Sample demonstrating the issue:**

```apex
public with sharing class SomeDummyClass {
    public static void someMethod() {
        if (!Schema.sObjectType.SomeCustomObject__c.isUpdateable()) {
            return;
        }

        SomeCustomObject__c someCustomObject = [
            SELECT Id, SomeCustomField__c
            FROM SomeCustomObject__c
            LIMIT 1
        ];

        if (someCustomObject.size() > 0) {
            update someCustomObject;
        }
    }

    public static void someOtherMethod() {
        SomeCustomObject__c someCustomObject = [
            SELECT Id, SomeCustomField__c
            FROM SomeCustomObject__c
            LIMIT 1
        ];

        if (someCustomObject.size() > 0) {
            update someCustomObject;
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 19 and line 26, but doesn't. This is a false-negative.
When removing the if at line 3, Apex PMD correctly reports the violations.

**Running PMD through:** *[CLI]*
